lib/src/package_meta.dart

```diff
@@ -432,7 +441,12 @@ class _FilePackageMeta extends PubPackageMeta {
   @override
   List<String> getInvalidReasons() {
     var reasons = <String>[];
-    if (_pubspec == null || _pubspec.isEmpty) {
+    if (_initializationErrors.isNotEmpty) {
+      reasons.addAll(_initializationErrors);
+    }
+    if (_pubspec == null) {
+      reasons.add('pubspec.yaml is corrupt');
```

Comment:
Is this message really necessary?  The only circumstance in which this could be reached is if an exception is caught in the call to `loadYaml`, in which case the exception message is already in `_initializationErrors`, so it seems redundant.